Framework/Algorithms/src/ConvertUnitsUsingDetectorTable.cpp

```diff
@@ -147,7 +148,7 @@ MatrixWorkspace_sptr ConvertUnitsUsingDetectorTable::convertViaTOF(
 
   // Perform Sanity Validation before creating workspace
   size_t checkIndex = 0;
-  int checkSpecNo = inputWS->getDetector(checkIndex)->getID();
+  int checkSpecNo = inputWS->getSpectrum(checkIndex).getSpectrumNo();
```

Comment:
Agreed. It will. From the Algorithm documentation, I'm led to believe that this should actually be the spectrum number.